### PR TITLE
Embeds: Add hex colours

### PR DESCRIPTION
**Pull request by dotVansh**

### CODE REVIEW

File: src/commands/moderation/message/Embed.ts [hunk 1]
```diff
@@ -150,10 +165,21 @@ export default class EmbedCommand extends BaseCommand {
 					return;
 				}
 
-				const embed = new EmbedBuilder()
+				let embed = new EmbedBuilder()
```
Comment (11xdeveloper) [inline src/commands/moderation/message/Embed.ts:168]: Not needed.

File: src/commands/moderation/message/Embed.ts [hunk 2]
```diff
@@ -150,10 +165,21 @@ export default class EmbedCommand extends BaseCommand {
...
 					.setTitle(embedTitle)
 					.setDescription(embedDescription);
 
+				if (embedColour && /^#?[\da-f]{6}$/i.test(embedColour)) try {
```
Comment (11xdeveloper) [inline src/commands/moderation/message/Embed.ts:172]: It's already a hex string, no need to have them include a hashtag.

File: src/commands/moderation/message/Embed.ts [hunk 3]
```diff
@@ -150,10 +165,21 @@ export default class EmbedCommand extends BaseCommand {
...
+					embed = embed.setColor(embedColour as HexColorString);
```
Comment (11xdeveloper) [inline src/commands/moderation/message/Embed.ts:173]: No need for `embed =`

File: src/commands/moderation/message/Embed.ts [hunk 4]
```diff
@@ -150,10 +165,21 @@ export default class EmbedCommand extends BaseCommand {
...
+				} finally { }
```
Comment (11xdeveloper) [inline src/commands/moderation/message/Embed.ts:174]: Remove the finally statement

File: src/commands/moderation/message/Embed.ts (hunk 3 above)
Comment (11xdeveloper) [inline src/commands/moderation/message/Embed.ts:173]: Remove the try statement

File: src/commands/moderation/message/Embed.ts [hunk 5]
```diff
@@ -150,10 +165,21 @@ export default class EmbedCommand extends BaseCommand {
...
+				else if (embedColour) {
+					await modalInteraction.reply({
```
Comment (11xdeveloper) [inline src/commands/moderation/message/Embed.ts:176]: Remove the await

**Review by 11xdeveloper — CHANGES_REQUESTED** [review]

File: src/commands/moderation/message/Embed.ts (hunk 2 above)
Comment (dotVansh) [reply src/commands/moderation/message/Embed.ts:172]: The hashtag is already optional in the regex test, tested

**Pull request merged**